tools/convert_tests/convert_tests_helper.py

```diff
@@ -0,0 +1,409 @@
+DQUOTE = "&"
+
+# Map timezone abbreviations to valid pytz timezone names
+timediff_abbr_map = {
+    "-0800": "US/Pacific",  # Pacific Standard Time
+    "-0500": "US/Eastern",  # Eastern Standard Time
+}
+
+timediff_zone_map = {
+    "-0800": "PST",  # Pacific Standard Time
+    "-0500": "EST",  # Eastern Standard Time
+}
+
+
+def convert_type(type_str, mapping):
+    """Helper function to convert a type string using the provided mapping."""
+    if type_str.startswith("list<") and type_str.endswith(">"):
+        # Extract the inner type and convert it
+        inner_type = type_str[5:-1]  # Extract what's inside "list<...>"
+        short_inner_type = mapping.get(inner_type.lower(), inner_type)
+        return f"list<{short_inner_type}>"
+
+    base_type, parameters = (type_str.split("<", 1) + [""])[:2]
+    parameters = f"<{parameters}" if parameters else ""
+    return mapping.get(base_type.lower(), base_type) + parameters
+
+
+def convert_to_long_type(type_str):
+    type_str = convert_type(type_str, long_name_map)
+    if type_str == "interval_year" or type_str == "interval_day":
+        type_str = "interval"
+    return type_str
+
+
+def match_sql_duration(value, value_type):
+    """
+    Check if a string is in the format of 'X days, HH:MM:SS'.
+    Returns a match object if successful, or None if it doesn't match.
+    """
+    if value_type != "str":
+        return None
+    pattern = r"^(?:(\d+)\s+days?,\s*)?(\d+):(\d+):(\d+)$"
+    return re.match(pattern, value)
+
+
+def format_timestamp(value):
+    """Format a timestamp value into ISO 8601 format, handling optional fractional seconds."""
+    # Try parsing with fractional seconds first
+    try:
+        dt = datetime.strptime(value, "%Y-%m-%d %H:%M:%S.%f")
+        return dt.isoformat()
+    except ValueError:
+        pass  # Continue to the next format if parsing fails
+
+    # Fallback to parsing without fractional seconds
+    try:
+        dt = datetime.strptime(value, "%Y-%m-%d %H:%M:%S")
+        return dt.isoformat()
+    except ValueError:
+        # Return the original value if it doesn't match expected formats
+        return value
+
+
+def has_last_colon_after_last_dash(s):
+    """
+    Check if the last ':' comes after the last '-' in the string.
+
+    Args:
+        s (str): Input string, e.g., "-10:30".
+
+    Returns:
+        bool: True if the last ':' comes after the last '-', False otherwise.
+    """
+    last_dash_index = s.rfind("-")  # Find the last occurrence of '-'
+    last_colon_index = s.rfind(":")  # Find the last occurrence of ':'
+
+    # Check if both '-' and ':' exist and if last ':' comes after last '-'
+    return (
+        last_dash_index != -1
+        and last_colon_index != -1
+        and last_dash_index < last_colon_index
+    )
+
+
+def format_timestamp_tz(timestamp_with_tz):
+    """Convert a timestamp with timezone abbreviation to ISO 8601 with offset."""
+    if "-" in timestamp_with_tz and not has_last_colon_after_last_dash(
```

Comment:
This line appears to be wanting a function to test whether the provided timestamp has a particular property such as being a timestamp with an explicit timezone in it.  Perhaps create a function for that purpose?